core/compute/compute_common.go

```diff
@@ -193,12 +193,17 @@ func setupVolumes(config *container.Config, instance model.Instance, hostConfig
 	if vMounts := instance.VolumesFromDataVolumeMounts; len(vMounts) > 0 {
 		for _, vMount := range vMounts {
 			storagePool := model.StoragePool{}
-			if ok, err := storage.IsVolumeActive(vMount, storagePool, client); !ok && err == nil {
+			// volume active == exists, possibly not attached to this host
+			if ok, err := storage.IsVolumeActive(vMount, storagePool, client); !ok {
```

Comment:
It's never a good idea to act on a return value before checking an `err`.  Having the first block be `!ok` is not safe.